segmentstore/server/src/main/java/io/pravega/segmentstore/server/logs/OperationProcessor.java

```diff
@@ -535,17 +537,15 @@ void commit(DataFrameBuilder.CommitArgs commitArgs) {
                     }
 
                     // Collect operations to commit.
-                    Timer memoryCommitTimer = new Timer();
                     toAck = collectCompletionCandidates(commitArgs);
 
                     // Commit metadata updates.
-                    int updateTxnCommitCount = OperationProcessor.this.metadataUpdater.commit(commitArgs.getMetadataTransactionId());
+                    OperationProcessor.this.metadataUpdater.commit(commitArgs.getMetadataTransactionId());
 
-                    // Commit operations to memory. Note that this will block synchronously if the Commit Queue is full (until it clears up).
+                    // Queue operations for memory commit, which will be done asynchronously.
                     toAck.forEach(OperationProcessor.this.commitQueue::add);
 
                     this.highestCommittedDataFrame = addressSequence;
-                    metrics.memoryCommit(updateTxnCommitCount, memoryCommitTimer.getElapsed());
```

Comment:
A couple of questions here:

1- Are you moving this to metrics update to line 419? What's the purpose of the change?
2- Can we test that the metrics are being computed correctly?

Reply:
1. This was measuring the correct thing until we moved the cache writes into an async background loop. At this time, this metric will always record 0 since the only thing done here is to commit metadata to memory, which is not very interesting.
2. I can't verify that this works "correctly" because there's no way I can measure the commit times from the outside, nor can I figure out the write batches that go through.
I moved this metric recording to the commit queue processor to measure (once again) what it was originally meant to record.